internal/api/core/routing.go

```diff
@@ -272,50 +227,51 @@ func allocateAPIRequest(endpoint string, apiHandler ValidAPIHandler) (any, *APIE
 }
 
 // Reflexively ensure that the api handler is of the correct type/format (e.g. looks like APIHandler).
+// Returns a ValidAPIHandler, the reflect.Type of the input struct and first output struct, and an APIError.
```

Comment:
You can be more clear here by writing more simply.
They can see the types in the signature, so you can leave them out.
Remember, documentation is not about repeating code.
It is about stating it in a different way.

So you can have something more clear that is like:
"Returns the handler, the type of ..., the type of ..., and an error.".